opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/jdbc/JdbcTracingTest.java

```diff
@@ -0,0 +1,141 @@
+    public void spanIsCreatedForPreparedStatement() throws SQLException {
+        PreparedStatement pstmt = dataSource.getConnection().prepareStatement("select 1");
+        assertTrue(pstmt.execute());
+        assertEquals(1, tracer.finishedSpans().size());
+    }
+
+    /**
+     * Make sure that a span is created when an active span exists joins the active
+     *
+     * @throws SQLException
+     */
+    @Test
+    public void spanJoinsActiveSpan() throws SQLException {
+        try (ActiveSpan ignored = tracer.buildSpan("parent").startActive()) {
+            assertTrue(dataSource.getConnection().prepareStatement("select 1").execute());
+            assertEquals(1, tracer.finishedSpans().size());
+        }
+
+        assertEquals(2, tracer.finishedSpans().size());
+        Optional<MockSpan> jdbcSpan = tracer
+                .finishedSpans()
+                .stream()
+                .filter((s) -> "java-jdbc".equals(s.tags().get(Tags.COMPONENT.getKey())))
+                .findFirst();
+
+        Optional<MockSpan> parentSpan = tracer
+                .finishedSpans()
+                .stream()
+                .filter((s) -> "parent".equals(s.operationName()))
+                .findFirst();
+
+        assertTrue(jdbcSpan.isPresent());
+        assertTrue(parentSpan.isPresent());
+
+        assertEquals(jdbcSpan.get().context().traceId(), parentSpan.get().context().traceId());
+        assertEquals(jdbcSpan.get().parentId(), parentSpan.get().context().spanId());
+    }
+
+    /**
+     * Make sure that a span is created when executing statements via a JDBC template
+     */
+    @Test
+    public void spanIsCreatedWhenUsingJdbcTemplate() {
+        jdbcTemplate.execute("select 1");
+        assertEquals(1, tracer.finishedSpans().size());
+    }
+
+    /**
+     * Sanity test for multiple requests, on their own threads, each executing a jdbc connection
+     */
+    @Test
+    public void concurrentParents() {
+        ExecutorService service = Executors.newFixedThreadPool(10);
+        IntStream.rangeClosed(1, 150).parallel().forEach(i -> {
+            service.submit(() -> {
+                // each iteration is like a request
+                try (ActiveSpan ignored = tracer.buildSpan("parent_" + i).startActive()) {
+                    jdbcTemplate.execute("select 1");
+                }
+            });
+        });
+
+        // for each request, we have one extra span, the jdbc one
+        await().until(() -> tracer.finishedSpans().size() == 300);
+
+        tracer.finishedSpans()
+                .stream()
+                .filter((s) -> s.operationName().startsWith("parent_"))
+                .forEach(parent -> {
+                    // each parent span should have one and only one child
+                    assertEquals(1,
+                            tracer.finishedSpans()
+                                    .stream()
+                                    .filter(child -> child.parentId() == parent.context().spanId())
```

Comment:
I think it also needs an assert like this:

`assertEquals(parentSpan.tags("statement"), childSpan.tags("statement"))` to ensure that the parent is correct. I am not sure what tags contains child span but there will be something which can be used to correlate with the parent  (in the for loop above it's necessary to set the statement on the parent and change it to `select + i`)

Reply:
Done, PR squashed.